tuf/src/client.rs

```diff
@@ -430,54 +429,34 @@ where
         }
     }
 
-    /// Returns the current trusted root version.
-    pub fn root_version(&self) -> u32 {
-        self.tuf.trusted_root().version()
+    /// Returns a reference to the TUF database.
+    pub fn database(&self) -> &Database<D> {
```

Comment:
Is there a concern with giving client holders access to potentially read untrusted metadata, mutate the Database, and mutate/inspect local/remote repos?  Sure, some caller constructed a tuf::Client from L/R originally, but this opens up the holder of a tuf::Client to access potentially invalid target metadata.

Reply:
Is your main concern `Client::database_mut()`? I could drop that since that's not something I really need at the moment.

That said, I'm not too worried about someone accessing that though for a few reasons:

* If you got a `&mut client`, you could always do `mem::swap(&mut client, &mut my_compromised_client)`.
* If you own the client, you could reform it with `Client::into_parts()` and `Client::from_parts(parts)`.
* Holders of `&mut client` could only inject signed metadata that meets the tuf schema. At that point, as far as the tuf spec is concerned the new metadata is valid. That could have been a complication if we tried to change the database during an update, but I've gotten rid of all the interior mutation, so we're protected from problems there by the borrow checker.

Reply:
I think of instances of tuf::Client as the capability to update metadata, lookup verified target descriptions, and read verified target data.  My main concern is that giving mutable field access can break any invariants we have between fields of Client or even within a field that wasn't intended to be publicly accessible.  For example, with `database_mut`, a user of Client could do a "partial update", calling only `update_targets` and never updating the root metadata, potentially trusting metadata that should be rejected by a key rotation that the Database never saw.  `Database` only implements the TUF spec when driven by `Client`.

I did not know of into_parts/from_parts.  Nevermind then, this doesn't expose any more mutable functionality than was already provided with Parts and mem::swap, and clients would have to go pretty far out of their way to fetch/parse unverified remote metadata through remote_repo()